### PR TITLE
SPSH-624 Implemented itslearning client

### PR DESCRIPTION
**Pull request by marode-cap**
# Description
Implements a client to interact with the ItsLearning IMS ES SOAP API

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: charts/dbildungs-iam-server/config/config.json [hunk 1]
```diff
@@ -45,5 +45,11 @@
     "KEYCLOAK_ADMINISTRATION_MODULE_LOG_LEVEL": "debug",
     "HEALTH_MODULE_LOG_LEVEL": "debug",
     "BACKEND_FOR_FRONTEND_MODULE_LOG_LEVEL": "debug"
+  },
+  "ITSLEARNING": {
+    "ENABLED": false,
+    "ENDPOINT": "https://localhost",
+    "USERNAME": "username",
```
Comment (kristoff-kiefer) [inline charts/dbildungs-iam-server/config/config.json:52]: hmmm, looks a little weird as a default, maybe https://itslearning.example.com would be better to show that it is indeed **not** a local service

File: config/config.json [hunk 2]
```diff
@@ -53,5 +53,11 @@
         "KEYCLOAK_ADMINISTRATION_MODULE_LOG_LEVEL": "debug",
         "HEALTH_MODULE_LOG_LEVEL": "debug",
         "BACKEND_FOR_FRONTEND_MODULE_LOG_LEVEL": "debug"
+    },
+    "ITSLEARNING": {
+        "ENABLED": false,
+        "ENDPOINT": "https://localhost",
```
Comment (kristoff-kiefer) [inline config/config.json:59]: Similar to above, the Demo-URL could and should be different maybe?

File: src/modules/itslearning/actions/base-action.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        };
+
+        Body: BodyResponse;
+    };
+};
+
+export abstract class IMSESAction<ResponseBodyType, ResultType> {
+    protected readonly xmlBuilder: XMLBuilder = new XMLBuilder({ ignoreAttributes: false });
+
+    protected readonly xmlParser: XMLParser = new XMLParser({
```
Comment (kristoff-kiefer) [inline src/modules/itslearning/actions/base-action.ts:30]: Think about injecting that maybe?

File: src/modules/itslearning/actions/read-all-persons.action.spec.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                pageIndex: 1,
+                pageSize: 10,
+            });
+
+            expect(action.buildRequest()).toBeDefined();
+        });
+    });
+
+    describe('isArrayOverride', () => {
+        it('should return true for specific tags', () => {
```
Comment (kristoff-kiefer) [inline src/modules/itslearning/actions/read-all-persons.action.spec.ts:17]: be more specific about "specific"

**Review by kristoff-kiefer — CHANGES_REQUESTED** [review]

**Review by kristoff-kiefer — APPROVED** [review]

File: src/modules/itslearning/actions/base-action.ts (hunk 3 above)
Comment (marode-cap) [reply src/modules/itslearning/actions/base-action.ts:30]: The parser may need specific configuration per action (currently only the override for 'isArray', which tells the parser if a tag should always be treated as a list, even if only one entry is present)

Injecting an already constructed parser would also mean you can't easily construct instances of the different actions

**Pull request merged**